src/xo-app/vm/tab-advanced.js

```diff
@@ -146,27 +144,23 @@ class CoresPerSocket extends Component {
   }
   render () {
     const vm = this.props.vm
-    const { formatMessage } = this.props.intl
 
     return <select
       className='form-control'
-      onChange={event => editVm(vm, { coresPerSocket: getEventValue(event) })}
-      value={vm.coresPerSocket}
+      onChange={event => editVm(vm, { coresPerSocket: +getEventValue(event) })}
+      value={vm.coresPerSocket || 'null'}
     >
-      <option value={'null'}> {formatMessage(messages.vmChooseCoresPerSocket)} </option>
-      {
-        map(
-          this._getCoresPerSocketPossibilities(),
-          coresPerSocket => <option
-            value={coresPerSocket}
-          >
-            {formatMessage(messages.vmCoresPerSocket, {
-              sockets: vm.CPUs.number / coresPerSocket,
-              value: coresPerSocket
-            })}
-          </option>
+      {_('vmChooseCoresPerSocket', message => <option value={'null'}>{message}</option>)}
```

Comment:
`{'null'}` can be written as `'null'` directly.

And you can simply use `''` for the empty value.